src/main/clojure/clara/rules/engine.cljc

```diff
@@ -1986,6 +1986,9 @@
     (let [query-node (get-in rulebase [:query-nodes query])]
       (when (= nil query-node)
         (platform/throw-error (str "The query " query " is invalid or not included in the rule base.")))
+      (when-not (= (into #{} (keys params)) ;; nil params should be equivalent to #{}
+                   (:param-keys query-node))
+        (platform/throw-error (str "The query " query " was not provided with the correct parameters, expected: " (:param-keys query-node) ", provided: " (set (keys params)))))
 
```

Comment:
Nit : Really long line?

Reply:
Looks like this was resolved by the changes in https://github.com/cerner/clara-rules/pull/455/commits/b559d93351f34cd058a05435b3a32dafa926bcb9